drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -23,6 +23,8 @@ LOG_MODULE_REGISTER(mcux_flexcomm);
 	COND_CODE_0(CONFIG_I2C_NXP_TRANSFER_TIMEOUT, (K_FOREVER),                                  \
 		    (K_MSEC(CONFIG_I2C_NXP_TRANSFER_TIMEOUT)))
 
+#define MCUX_FLEXCOMM_MAX_TARGETS 4
```

Comment:
I guess ideally this would be part of device tree

Reply:
> I guess ideally this would be part of device tree

I could add something like `num-targets` or `max-targets` to the lpc-i2c?

Reply:
Does the number of target devices differ between the SoCs using this driver? If not, why put it in the device tree?

Reply:
I checked the LPC55xxx, LPC54xxx datasheets and the hal_nxp and they all seem to refer to 4 target addresses.

Reply:
@DerekSnell are you aware of any instances of the Flexcomm IP that support more than 4 target addresses? The RT595 also seems to only support 4

Reply:
Ah then yeah this seems ok. The naming of this, without looking at the IP suggested it was a maximum and there may be smaller numbers for some IP instances perhaps. Ignore this DT comment in that case.